loopy/symbolic.py

```diff
@@ -2030,32 +2055,41 @@ def map_subscript(self, expr, inames):
         if expr.aggregate.name not in self.var_names:
             return
 
+        if expr.aggregate.name in self.bad_subscripts:
+            return
+
         arg_name = expr.aggregate.name
         subscript = expr.index_tuple
 
         descriptor = self.kernel.get_var_descriptor(arg_name)
 
         try:
-            access_range = get_access_range(
+            access_map = get_access_map(
                     domain, subscript, self.kernel.assumptions,
                     shape=descriptor.shape if self.overestimate else None,
                     allowed_constant_names=self.kernel.get_unwritten_value_args())
         except UnableToDetermineAccessRange:
             self.bad_subscripts[arg_name].append(expr)
             return
 
-        if self.access_ranges[arg_name] is None:
-            self.access_ranges[arg_name] = access_range
-        else:
-            if (self.access_ranges[arg_name].dim(dim_type.set)
-                    != access_range.dim(dim_type.set)):
+        # {{{ check that access dimensionality is the same.
```

Comment:
the same as what?

Reply:
Corrected in 4cbefe2dcd5bc9c5f9515b69b37c60cf32dc3278